Sources/SentrySwiftUI/SentryTracedView.swift

```diff
@@ -34,41 +34,61 @@ import SentryInternal
 ///         //The part of your content you want to measure
 ///     }.sentryTrace("My Awesome Screen")
 ///
-///
 @available(iOS 13, macOS 10.15, tvOS 13, watchOS 6.0, *)
 public struct SentryTracedView<Content: View>: View {
-    
+    @State var viewAppeared = false
+
     let content: () -> Content
     let name: String
-    let id: SpanId
-    
+    let nameSource: SentryTransactionNameSource
+
     public init(_ viewName: String? = nil, content: @escaping () -> Content) {
         self.content = content
         self.name = viewName ?? SentryTracedView.extractName(content: Content.self)
-        id = SentryPerformanceTracker.shared.startSpan(withName: self.name,
-                                                       nameSource: viewName == nil ? .component : .custom,
-                                                       operation: "ui.load")
+        self.nameSource = viewName == nil ? .component : .custom
     }
-    
+
     private static func extractName(content: Any) -> String {
         var result = String(describing: content)
-        
+
         if let index = result.firstIndex(of: "<") {
             result = String(result[result.startIndex ..< index])
         }
-        
+
         return result
     }
-    
+
     public var body: some View {
+        if viewAppeared {
+            return self.content().onAppear()
+        }
+
+        var transactionCreated = false
+        if SentryPerformanceTracker.shared.activeSpanId() == nil {
+            transactionCreated = true
+            let transactionId = SentryPerformanceTracker.shared.startSpan(withName: self.name, nameSource: nameSource, operation: "ui.load")
+            SentryPerformanceTracker.shared.pushActiveSpan(transactionId)
+
+            //According to Apple's documentation, the call to `body` needs to be fast
+            //and can be made many times in one frame. Therefore they don't use async code to process the view.
+            //Scheduling to finish the transaction at the end of the main loop seems the least hack solution right now.
+            DispatchQueue.main.async {
+                SentryPerformanceTracker.shared.popActiveSpan()
+                SentryPerformanceTracker.shared.finishSpan(transactionId)
+            }
+        }
+
+        let id = SentryPerformanceTracker.shared.startSpan(withName: transactionCreated ? "\(self.name).body" : self.name, nameSource: nameSource, operation: "ui.load")
```

Comment:
`m`: I think `.body` looks a bit weird. Maybe the following is better
```suggestion
        let id = SentryPerformanceTracker.shared.startSpan(withName: transactionCreated ? "\(self.name) - body" : self.name, nameSource: nameSource, operation: "ui.load")
```

<img width="432" alt="Screenshot 2023-02-13 at 15 31 07" src="https://user-images.githubusercontent.com/2443292/218485695-61859bef-9a46-46cd-b77b-4b119f677ee5.png">